src/unix/solarish/mod.rs

```diff
@@ -1361,7 +1312,7 @@ pub const RLIMIT_AS: ::c_int = RLIMIT_VMEM;
 
 #[deprecated(since = "0.2.64", note = "Not stable across OS versions")]
 pub const RLIM_NLIMITS: rlim_t = 7;
-pub const RLIM_INFINITY: rlim_t = 0x7fffffff;
+pub const RLIM_INFINITY: rlim_t = 0xfffffffffffffffd;
```

Comment:
This was previously using the 32-bit value.  Given that all of the illumos/Solaris targets are 64-bit today, it made sense to use that value.

Reply:
Wouldn’t it be a good idea to make this definition different for 32 and 64 bit targets ?

Reply:
The existing builds for rust on both Solaris and illumos target only x86_64.  If/when someone is inclined to set up a 32-bit build, they'll need to check all of the libc interfaces, this constant included, for conformance.